docs/operator-manual/dynamic-cluster-distribution.md

```diff
@@ -0,0 +1,283 @@
+# Dynamic Cluster Distribution
+
+Sharding in Argo CD uses StatefulSet for the application controller. Although the application controller does not have any state to preserve, StatefulSets are used to get predictable hostnames and the serial number in the hostname is used to get the shard id of a particular instance.
+
+But using StatefulSet has the following limitations:
+
+* Any change done to the StatefulSet would cause all the child pods to restart in a serial fashion. This makes scaling up/down of the application controller slow as even existing healthy instances need to be restarted as well. 
+
+* Each shard replica knows about the total number of available shards by evaluating the environment variable ARGOCD_CONTROLLER_REPLICAS, which needs to be kept up-to-date with the actual number of available replicas. If the number of replicas in the StatefulSet does not equal the number set in environment variable ARGOCD_CONTROLLER_REPLICAS, sharding will not work as intended, leading to both, unused and overused replicas. As this environment variable is set on the StatefulSet and propagated to the pods, all the pods in the StatefulSet need to be restarted in order to pick up the new number of total shards.
+
+
+Starting v2.9, ArgoCD supports a dynamic cluster distribution feature. In this mechanism, Argo CD sharding uses Deployments for the application controller. 
+
+
+## Enabling Dynamic Distribution of Clusters
+
+Inorder to utilize the feature, the StatefulSet of Application Controller needs to be replaced with the Deployment Configuration of Application Controller. To do so, set the number of replicas of StatefulSet and the environment variable `ARGOCD_CONTROLLER_REPLICAS` to 0 disabling all current application controllers.
+
+Once all the current controllers have stopped working, apply the below deployment configuration of application controller with the desired number of replicas.
+
+```yaml
```

Comment:
I think it is better to create a dedicated overlay in the `manifests` folder to provide this new deployment. Later on we could promote this overlay to the base. WDYT?

Reply:
Good idea.